roadmap/implementers-guide/src/node/disputes/participation.md

```diff
@@ -0,0 +1,106 @@
+# Dispute Participation
+
+Tracks all open disputes in the active leaves set.
+
+## Remarks
+
+`UnsignedTransaction`s are OK to be used, since the inner
+element `CommittedCandidateReceipt` is verifiable.
+
+## Assumptions
+
+* If each validator gossips its own vote regarding the disputed block, upon receiving the first vote (the initiating dispute vote) there is only
+a need for a very small gossip only containing the `Vote` itself.
```

Comment:
I've trouble parsing this: 

I presume you are not saying they should wait to gossip the dispute until they vote?  That'd take too long.  

Are you saying politeness is simple because they could assume their gossip partners got their previous gossiped message about the dispute?  Yes this seems reasonable.  We could've some churn in the gossip pattern though, so maybe one wants a way to update this later, although probably not anytime soon.  I'm unsure how robust politeness is on this sort of point elsewhere.

Reply:
I am not saying they should not gossip _other_ votes of other validators, but a validator should cast its own vote upon reception of the first validity challenging vote.
I added the vote definition of a `Vote` here which was missing previously https://github.com/paritytech/polkadot/pull/2192/files#diff-87b3e890a1e89a1ec02e48556fa509020e0ca6c88cd447e3de9ca84f57af9647R59-R85

Reply:
Yes fine, they do begin their checking process whenever they discover a challenge.

We mostly keep our gossip protocols "polite" in that nodes track one anothers' knowledge and does not forward messages until we know the recipient possesses the precursor knowledge.  It's a higher level back pressure mechanism from a developers perspective that reliably bounds network bandwidth and avoids messy internal queuing or resequencing logic for handling out-of-context messages.  We design complex protocols more easily using politeness.  In fact, politeness requires attention to protocol gossip assumptions, like around messages delays, but so far it never proved too tricky there.

Reply:
> We mostly keep our gossip protocols "polite" in that nodes track one anothers' knowledge and does not forward messages until we know the recipient possesses the precursor knowledge. It's a higher level back pressure mechanism from a developers perspective that reliably bounds network bandwidth and avoids messy internal queuing or resequencing logic for handling out-of-context messages. We design complex protocols more easily using politeness. In fact, politeness requires attention to protocol gossip assumptions, like around messages delays, but so far it never proved too tricky there.

That makes sense and I did not associate that with politeness :) but rather "not-spam". Essentially this mimics the availability distribution pretty well.

Reply:
I think politeness provides an extremely useful design pattern that simplifies the interaction between research and development, but that said...

We've considered non-polite options for announcing approval assignment VRFs as possible defenses against highly targeted DoS attacks on soundness, which ideally want all validators to learn the assignments whenever malicious ones do. 
1. Assignment announcements could be sent directly to everyone in single UDP packets, which perhaps precludes knowing that everyone saw the relay chain block, and definitely creates hiccups in the gossip protocol.  
2. An erasure coded reliable gossip protocol could ensure assignment announcements reached all validators if they reached malicious ones, which maybe could ensure knowledge, but gets noisy for such small messages, and requires slashing to prevent false message spam.
3. If we adopted the ring VRF here, then a mix network could transport assignment announcements with less noise, but adds latency, damages liveness, and is not as polite.

We might adopt 2 or 3 one day, which remain somewhat polite if messy, but that's a problem for another day.

Reply:
I figure we can probably just piggyback on the existing transaction gossip that exists in Substrate. If validators create regular old operational Unsigned transactions, then other nodes will gossip and include those. No need for additional messaging.